test/cpsubsystem/AtomicLongProxyTest.js

```diff
@@ -0,0 +1,162 @@
+    });
+
+    it('should create AtomicLong with respect to given CP group', async function () {
+        const longInAnotherGroup = await client.getAtomicLong('along@mygroup');
+
+        const value1 = await longInAnotherGroup.incrementAndGet();
+        expectLong(1, value1);
+        // the following value has to be 0,
+        // as `long` belongs to the default CP group
+        const value2 = await long.get();
+        expectLong(0, value2);
+    });
+
+    it('destroy: should destroy AtomicLong', async function () {
+        const anotherLong = await client.getAtomicLong('another-long');
+        await anotherLong.destroy();
+
+        try {
+            await anotherLong.get();
+        } catch (err) {
+            expect(err).to.be.instanceOf(DistributedObjectDestroyedError);
+        }
+    });
+
+    it('get: should return 0 initially', async function () {
+        const value = await long.get();
+        expectLong(0, value);
+    });
+
+    it('addAndGet: should return added value', async function () {
+        const value = await long.addAndGet(33);
+        expectLong(33, value);
+    });
+
+    it('addAndGet: should add', async function () {
+        await long.addAndGet(33);
+        const value = await long.get();
+        expectLong(33, value);
+    });
+
+    it('getAndAdd: should return old value', async function () {
+        const value = await long.getAndAdd(123);
+        expectLong(0, value);
+    });
+
+    it('getAndAdd: should add', async function () {
+        await long.getAndAdd(123);
+        const value = await long.get();
+        return expectLong(123, value);
+    });
+
+    it('decrementAndGet: should decrement', async function () {
+        const value = await long.decrementAndGet();
+        expectLong(-1, value);
+    });
+
+    it('compareAndSet: should set the value when condition is met', async function () {
+        await long.set(42);
+        const result = await long.compareAndSet(42, 13);
+        expect(result).to.be.true;
+        const value = await long.get();
+        expectLong(13, value);
+    });
+
+    it('compareAndSet: should have no effect when condition is not met', async function () {
+        await long.set(42);
+        const result = await long.compareAndSet(13, 13);
+        expect(result).to.be.false;
+        const value = await long.get();
+        expectLong(42, value);
+    });
+
+    it('set: should set new value', async function () {
+        await long.set(1001);
+        const value = await long.get();
+        expectLong(1001, value);
+    });
+
+    it('getAndSet: should return old value', async function () {
+        const value = await long.getAndSet(-123);
+        expectLong(0, value);
+    });
+
+    it('getAndSet: should set the value', async function () {
+        await long.getAndSet(-123);
+        const value = await long.get();
+        expectLong(-123, value);
+    });
+
+    it('incrementAndGet: should increment', async function () {
```

Comment:
Same comment with `decrementAndGet: should decrement`

Reply:
See my comment for `decrementAndGet`